builtin/providers/aws/resource_aws_iam_user.go

```diff
@@ -132,39 +138,25 @@ func resourceAwsIamUserUpdate(d *schema.ResourceData, meta interface{}) error {
 	}
 	return nil
 }
+
 func resourceAwsIamUserDelete(d *schema.ResourceData, meta interface{}) error {
 	iamconn := meta.(*AWSClient).iamconn
 
 	// IAM Users must be removed from all groups before they can be deleted
 	var groups []string
-	var marker *string
-	truncated := aws.Bool(true)
-
-	for *truncated == true {
-		listOpts := iam.ListGroupsForUserInput{
-			UserName: aws.String(d.Id()),
-		}
-
-		if marker != nil {
-			listOpts.Marker = marker
-		}
-
-		r, err := iamconn.ListGroupsForUser(&listOpts)
-		if err != nil {
-			return err
-		}
-
-		for _, g := range r.Groups {
+	listGroups := &iam.ListGroupsForUserInput{
+		UserName: aws.String(d.Id()),
+	}
+	pageOfGroups := func(page *iam.ListGroupsForUserOutput, lastPage bool) (shouldContinue bool) {
+		for _, g := range page.Groups {
 			groups = append(groups, *g.GroupName)
 		}
-
-		// if there's a marker present, we need to save it for pagination
-		if r.Marker != nil {
-			*marker = *r.Marker
-		}
-		*truncated = *r.IsTruncated
+		return true
```

Comment:
Isn't this going to loop forever if `shouldContinue = true`? Shouldn't this be something like

```go
if !lastPage {
  return true
}
```

The previous implementation used `ListGroupsForUserOutput.IsTruncated` to make decisions when paginating. I don't mind changing the implementation as long as the behaviour remains the same. Can you confirm the behaviour is exactly the same?